src/backend/catalog/catalog.c

```diff
@@ -362,9 +362,20 @@ IsExtAuxNamespace(Oid namespaceId)
  *		system objects only.  As of 8.0, this was only true for
  *		schema and tablespace names.  With 9.6, this is also true
  *		for roles.
+ */
+bool
+IsReservedName(const char *name)
+{
+	/* ugly coding for speed */
```

Comment:
need to check name length ?

Reply:
It looks like PG guys write this ugly code for speed and it is safe for shorter length name.